UploadServlet.java

```diff
@@ -72,86 +114,96 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
-				
-				ps = con.prepareStatement(sql);
-				
-				ps.setString(1, subject);
-				ps.setString(2, title);
-				ps.setString(3, highlights);
-				ps.setString(4, abstracts);
-				ps.setString(5, author);
-				ps.setTimestamp(6, time);
-				ps.setString(7, fileName);
-				ps.setString(8, path);
-				
-				int status = ps.executeUpdate();
+				try
+				{
+					Class.forName("com.mysql.jdbc.Driver");
+					
+					String u = "root";
+					String password = "124536";
+					String ul = "jdbc:mysql://localhost:3306/ooad?autoReconnect=true&&useSSL=false";
+					
+					con = DriverManager.getConnection(ul, u, password);
+					
+					String sql = "insert into article(subject, title, highlight, abstracts, author, time, filename, path) values (?,?,?,?,?,?,?,?)";
+					
+					ps = con.prepareStatement(sql);
+					
+					ps.setString(1, subject);
+					ps.setString(2, title);
+					ps.setString(3, highlights);
+					ps.setString(4, abstracts);
+					ps.setString(5, author);
+					ps.setTimestamp(6, time);
+					ps.setString(7, fileName);
+					ps.setString(8, path);
+					
+					int status = ps.executeUpdate();
+					
+					if(status>0)
+					{
+						session.setAttribute("fileName", fileName);
+						String msg = "" +fileName+ " upload successfully";
+						request.setAttribute("msg", msg);
+						request.setAttribute("subject", subject);
+						request.getRequestDispatcher("NewFile.jsp").forward(request, response);
+
+					}
+							
+			}
 				
-				if(status>0)
+				catch(SQLException | ClassNotFoundException e)
 				{
-					session.setAttribute("fileName", fileName);
-					String msg = "" +fileName+ " upload successfully";
-					request.setAttribute("msg", msg);
-					request.setAttribute("subject", subject);
-					request.getRequestDispatcher("NewFile.jsp").forward(request, response);
-//					System.out.println("uploadpath: "+uploadPath);
+					System.out.println(e);
 				}
 				
-			}
-			
-			catch(SQLException | ClassNotFoundException e)
-			{
-				System.out.println(e);
-			}
-			
-			finally
-			{
-				try
+				finally
 				{
-					if(ps != null)
+					try
 					{
-						ps.close();
+						if(ps != null)
+						{
+							ps.close();
+						}
+						
+						if(con!=null)
+						{
+							con.close();
+						}
 					}
 					
-					if(con!=null)
+					catch(SQLException e)
 					{
-						con.close();
+						out.println(e);
 					}
 				}
 				
-				catch(SQLException e)
-				{
-					out.println(e);
-				}
+			}else {// if the uploaded file type is not PDF, show an alert message
+				
+				JOptionPane.showMessageDialog(null, "Unsupported article type!.\nOnly PDF files are supported. ", "Info", JOptionPane.INFORMATION_MESSAGE);
+				
+				request.setAttribute("author", author);
+				request.setAttribute("subject", subject);
+				request.setAttribute("title", title);
+				request.setAttribute("highlights", highlights);
+				request.setAttribute("abstracts", abstracts);
+				request.getRequestDispatcher("PostArticle.jsp").forward(request, response);

```

Comment:
@hema-001 
The above block of statements have appeared on lines 87-92.

This calls for a refactoring effort.  For example, you could write  a helper function for setting attributes.

**Hui**

Reply:
For now, I think it is unnecessary to do so.

Because, we would like to minimize the amount of components in the system to preserve minimal complexity.

It would be kind of pointless to create a helper that we are going to use only one time in one place.

I don't think DRY principle applies here, since we are setting attributes twice only(I did repeat it the second time, originally the attributes were set once only).

Thanks.

Reply:
> For now, I think it is unnecessary to do so.
> 
> Because, we would like to minimize the amount of components in the system to preserve minimal complexity.
> 
> It would be kind of pointless to create a helper that we are going to use only one time in one place.
> 
> I don't think DRY principle applies here, since we are setting attributes twice only(I did repeat it the second time, originally the attributes were set once only).
> 
> Thanks.

OK.

Hui